test/libsolidity/syntaxTests/dataLocations/externalFunction/function_argument_location_specifier_test_external_transient_storage.sol

```diff
@@ -0,0 +1,5 @@
+contract test {
+	function f(bytes transient) external;
```

Comment:
You are right, this was meant to be data location. Since `transient` is not a keyword, it probably is read as the variable name (although I should check that).